### PR TITLE
Ecdsa cross compilation fix + api change

### PR DESCRIPTION
**Pull request by jakubDoka**
Dude to the `std.crypto.ecdsa.KeyPair.create` taking and optional of seed, even if the seed is generated, cross-compiling to the environments without standard random source (eg. wasm) (`std.crypto.random.bytes`) will fail to compile.

This PR changes the API of the problematic function and moves the random seed generation to a new utility function.

The simplest workaround is copying the ECDSA file to my project and making the changes (to enjoy microscopic crypto wasm binary).

### CODE REVIEW

**Comment by andrewrk** [comment]
Thank you, this is a much better API.

File: lib/std/os/linux.zig [hunk 1]
```diff
@@ -7175,7 +7175,7 @@ pub const perf_event_attr = extern struct {
     /// Defines size of the user stack to dump on samples.
     sample_stack_user: u32 = 0,
 
-    clockid: clockid_t = .REALTIME,
+    clockid: clockid_t = 0,
```
Comment (jedisct1) [inline lib/std/os/linux.zig]: Was that meant to be part of the changes?

Comment (alexrp) [reply lib/std/os/linux.zig]: This seems to be an old commit that wasn't included in the merged PR.

Comment (jakubDoka) [reply lib/std/os/linux.zig]: Yes, I noticed it in diff and removed the change

**Comment by jakubDoka** [comment]
> We should do the same for `ed25519.KeyPair` and MLKEM/Kyber for consistency.
> 
> `create` and `generate` are synonyms. To make the function name describe the intent, how about renaming `generate()` to `createRandom()`, or `create()` to `createDeterministic()`?

I don't mind making a PR, assuming naming is decided.

**Comment by andrewrk** [comment]
@jedisct1 `create` typically means it will heap-allocate, and there should be a corresponding `destroy`.

Instead, the pure function (the one that takes the seed as a parameter) should be called `init`, and the convenience method, which arguably should not even exist, should be called the awkward name, such as `initWithRandomSeed`.

**Pull request merged**